### PR TITLE
FFC-135605: create an openAPI for eligibility endpoint

### PR DESCRIPTION
**Pull request by marcinmo-kainos**

### CODE REVIEW

**Review by liamwilson-kainos — APPROVED** [review]
I think this is all fine to me! 

Maybe we can also find out about whether we need to secure the API in any way, is it okay to keep it unauthenticated as it will be an internal call from one node server to the other inside the Kubernetes cluster?

Do Defra have any patterns around the api path that is exposed, e..g with a version prefix `/v1` or something like that?

Could even as part of this ticket define the API route and get ahead of the plan

File: app/server.js [hunk 1]
```diff
@@ -7,7 +7,8 @@ const server = Hapi.server({
 
 const routes = [].concat(
   require('./routes/healthy'),
-  require('./routes/healthz')
+  require('./routes/healthz'),
+  require('./routes/api/eligibility')
```
Comment (liamwilson-kainos) [inline app/server.js:11]: Maybe following our pattern we could move this route under the new auto-eligibility folder on my branch?

File: openApi/openApi.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
+openapi: 3.0.3
```
Comment (liamwilson-kainos) [inline openApi/openApi.yaml:1]: I added a `docs` folder on my branch with the asyncapi spec as well. Can maybe move this there once rebased? :)

File: package.json [hunk 3]
```diff
@@ -19,6 +19,7 @@
   ],
   "license": "OGL-UK-3.0",
```
Comment (liamwilson-kainos) [inline package.json:20]: Don't forget to bump the semantic version number

**Review by liamwilson-kainos — APPROVED** [review]

File: app/server.js (hunk 1 above)
Comment (marcinmo-kainos) [reply app/server.js:11]: agree, i'll realign with your changes

File: package.json (hunk 3 above)
Comment (marcinmo-kainos) [reply package.json:20]: bumped

File: openApi/openApi.yaml (hunk 2 above)
Comment (marcinmo-kainos) [reply openApi/openApi.yaml]: moved to the docs folder

File: app/register-your-interest/routes/eligibility.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+const Joi = require('joi')
+const Boom = require('@hapi/boom')
```
Comment (liamwilson-kainos) [inline app/register-your-interest/routes/eligibility.js:2]: should we put this under the `app/auto-eligibility folder`?

Comment (marcinmo-kainos) [reply app/register-your-interest/routes/eligibility.js]: done

**Review by liamwilson-kainos — APPROVED** [review]

**Pull request merged**